src/wp-includes/post.php

```diff
@@ -1143,7 +1143,9 @@ function get_post_mime_type( $post = null ) {
  * @return string|false Post status on success, false on failure.
  */
 function get_post_status( $post = null ) {
-	$post = get_post( $post );
+	if ( ! is_object( $post ) ) {
```

Comment:
In my opinion, this change is a step in the right direction.
It doesn't make sense to retrieve an instance of `WP_Object` if it's already loaded. This also brings some performance improvement.
After examining the GitHub log, it appears there's no particular reason to call
```php
$post = get_post( $post );
```
every time the `get_post_status()` function is called.

Reply:
Hmm, not so sure as `$post = get_post( $post );` deals with caching (all different kinds of it). removing it might bring some unexplained regressions elsewhere.

Reply:
>  Seems removing it might bring some hard to explain/find regressions.

That was also my concern, @azaozz.
Some code might assume that `get_post($post)` is called even when there's objectively no need to (because `$post` is already an object).
However, I see the change that @youknowriad proposed as an improvement, but I also agree that it **_might_** lead to some hard to find regressions.

Reply:
> objectively no need to (because $post is already an object).

Right, it may be an object but it may not be synced with the cache? Like cache may need to be updated, or (external) cache has changes in the meantime, or some external process/DB connection runs slow at the moment and this might be affected for some reason, or... who knows what else :)

Reply:
`$post = get_post( $post )` is a very common pattern in WP-dev to normalize the post object. A regex search for `\$post\s*= get_post\( \$post \)` returns 121 results in `src` and probably misses a few minor variations.

I know it normalizes both integers and `WP_Post` objects but it may help with arrays too (Ozz, you probably have more intimate knowledge here).

If it's a pattern that could be improved, then it's way outside the scope of this PR.